sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/remote_cryptography_client.hpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A remote client used to perform cryptographic operations with Azure Key Vault keys.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/response.hpp>
+#include <azure/core/url.hpp>
+
+#include <azure/keyvault/common/internal/keyvault_pipeline.hpp>
+
+#include "azure/keyvault/keys/cryptography/cryptography_client_options.hpp"
+#include "azure/keyvault/keys/cryptography/cryptography_provider.hpp"
+#include "azure/keyvault/keys/cryptography/encrypt_parameters.hpp"
+#include "azure/keyvault/keys/cryptography/encrypt_result.hpp"
+#include "azure/keyvault/keys/key_vault_key.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure {
+  namespace Security {
+    namespace KeyVault {
+      namespace Keys {
+        namespace Cryptography {
+  namespace _detail {
+
+    struct RemoteCryptographyClient
```

Comment:
The RemoteClient is an implementation detail.  It is not exposed to end-customers.

It uses public Pipeline because the same pipeline is shared between the crypto client and the remote client